improve JS example

Hades32 opened the pull request.

danthegoodman1 commented on docs/integration/js-example.md:
```diff
@@ -1,9 +1,154 @@
+
+## Integrating the game into the Ultimate Arcade
+
+### Dockerfile for your server
```
Let's point them to the one in the example?

Hades32 replied:
Why not to the one included in the SDK? I expect that we'll forget to update the example after a while but hopefully not the SDK

danthegoodman1 replied:
Fair

danthegoodman1 commented on docs/integration/js-example.md:
```diff
@@ -1,9 +1,154 @@
+
+On the frontend run `npm i --save @ultimatearcade/client-sdk` and on the backend run `npm i --save @ultimatearcade/server-sdk`.
+
+BTW, you can use a single code base for frontend and backend, but you need to ship us a folder with static assets (HTML,JS,CSS,...) for the frontend as we don't run the frontend as a nodeJS server to save resources.
```
```suggestion
```
Lets reduce text where possible, this could also confuse them

danthegoodman1 commented on docs/integration/js-example.md:
````diff
@@ -1,9 +1,154 @@
+const = si.player_token
+```
+
+### Ensuring the connected player is valid
````
```suggestion
### Activate the player's game session
```

danthegoodman1 commented on docs/integration/js-example.md:
```diff
@@ -1,9 +1,154 @@
+
+### Ensuring the connected player is valid
+
+When you get a connection (probably via WebSockets) you still need to ensure that a legitimate user has reached your server.
```
```suggestion
```
Reducing text

danthegoodman1 commented on docs/integration/js-example.md:
````diff
@@ -1,9 +1,154 @@
+const playerID = actResp.player_id
+```
+
+You MUST use the playerID from this call to identify the player to ensure that the same player can't connect twice. We try to prevent this on our end, too, but there are always some small time windows when this might happen.
````
```suggestion
You MUST use the playerID from this call to identify the player to ensure that the same player can't connect twice.
```

danthegoodman1 commented on docs/integration/js-example.md:
```diff
@@ -1,9 +1,154 @@
+
+You should also not call this method twice. So it's a good idea to store the `player_token` in your player object, and if you get a connection check if it's just a reconnect after a dropped connection by comparing the player tokens.
+
+### Starting the game, locking the pool
```
```suggestion
### (Pooled games) Starting the game
```

We might also want to show how to do it for leaderboard games, maybe another tutorial

danthegoodman1 commented on docs/integration/js-example.md:
```diff
@@ -1,9 +1,154 @@
+
+#### Ending the game with a winner
+
+:::note
+This is only relevant for pool games
+:::
```
If this is a pool game tutorial, and we've clearly stated that, then we can remove these

danthegoodman1 replied:
we'll be pretty hands on with integrations anyway

Hades32 replied:
hmm, true, TBH I forgot that we already did basically the same thing in the code-adaptions... I'll think how to unify or better differentiate the two

Hades32 commented:
@danthegoodman1 I made the tutorial more focused on the actual example and moved over the extra info I had there into the code-adaptions page. I also removed the "don't do real-time" stuff and replaced it with latency-sensitive vs not.

Feel free to make any changes as needed

danthegoodman1 commented on docs/integration/js-example.md:
```diff
@@ -1,9 +1,143 @@
+sidebar_position: 8
 ---
 
 # Javascript Example Game Integration
```
```suggestion
# JavaScript Example Game Integration
```

danthegoodman1 commented on docs/supported-game-types/real-time.md:
```diff
@@ -4,28 +4,25 @@ sidebar_position: 2
 # Real-time Multiplayer
 
-While real-time multiplayer games are possible (like [cyberworms](https://ultimatearcade.io/cyber-worms) and [cyberworms battle royale](https://ultimatearcade.io/cyber-worms?mode=battle-royale)), they can be more difficult to build, integrate, maintain, and monetize.
+While in async-multiplayer games you interact with your opponents only indirectly (e.g. competing for a better score) in real-time multiplayer games directly interact with them.
```
```suggestion
While in async-multiplayer games you interact with your opponents only indirectly (e.g. competing for a better score) in real-time multiplayer games players directly interact with each other.
```

danthegoodman1 commented on docs/supported-game-types/real-time.md:
```diff
@@ -4,28 +4,25 @@ sidebar_position: 2
+
+If your game doesn't mind if there's a few hundred milliseconds delay, then it's far easier to host them and keep up a great player experience.
+
+For real-time games we support to types of games
```
```suggestion
For real-time games we support two types of games:
```

danthegoodman1 commented on docs/supported-game-types/real-time.md:
```diff
@@ -4,28 +4,25 @@ sidebar_position: 2
+If your game doesn't mind if there's a few hundred milliseconds delay, then it's far easier to host them and keep up a great player experience.
+
+For real-time games we support to types of games
+* *Player vs Player (PvP)* - players come and go and might fight each other in an open session
```
```suggestion
* *Player vs Player (PvP)* - players come and go, might fight each other in an open session
```

danthegoodman1 reviewed (changes requested):
sorry I didn't make the above adjustments part of the review!

danthegoodman1 commented:
just did it myself lol
